### ISSUE TITLE
[16/1439] Generating rom_ext_signer with a custom command error: no such subcommand: `+`

### ISSUE THREAD

**Issue by nicolast0604** [lowRISC/opentitan#6654]
Hi 
I am trying to "Create an FPGA bitstream" but met the following error.
Is there anyone have the same case? 
Please help if you have any hint, thanks a  lot.

```shell
~/opentitan$ ninja -C build-out all
ninja: Entering directory `build-out'
[16/1439] Generating rom_ext_signer with a custom command
error: no such subcommand: `+`

	Did you mean `doc`?
[17/1439] Compiling C++ object 'sw/device/lib/base/testing/acf5797@@mock_mmio@sta/mock_mmio.cc.o'
FAILED: sw/host/rom_ext_image_tools/signer 
/home/nioclast0604/opentitan/util/invoke_cargo.sh /usr/bin/cargo '--manifest-path=/home/nioclast0604/opentitan/sw/host/rom_ext_image_tools/signer/Cargo.toml --target-dir=/home/nioclast0604/opentitan/build-out/sw/host/rom_ext_image_tools/signer --release ' '' '' /home/nioclast0604/opentitan /home/nioclast0604/opentitan/build-out
[19/1439] Compiling C++ object 'sw/device/lib/base/testing/acf5797@@mock_mmio@sta/mock_mmio.cc.o'
```

**Comment by moidx** [comment]
Copying this from @alphan:

#6524 added new dependencies for ROM_EXT image signer so that it can be built by default. The following dependencies will be required for a successful build:
- Cmake (2.8.11 or later)
- Go (1.10 or later)
- Perl (5.6.1 or later)
- Rust (1.52.1)

You can install Cmake, Go, and Perl using apt on debian or ubuntu. For Rust, if you have a previous version installed, you can use rustup. If not, you can use the vendored rustup-init.sh script (`sw/vendor/rustup/rustup-init.sh -y --default-toolchain 1.52.1`) once #6524 is merged, or follow the instructions at https://www.rust-lang.org/learn/get-started.

**Comment by moidx** [comment]
The getting started documentation can probably be updated to help clarify the Rust instruction steps.

**Comment by nicolast0604** [comment]
Thanks moidx
It seems works however I have another issue as below:


[96/1485] Generating 'sw/device/lib/dif/libsw_lib_dif_alert_handler.a.p/alert_handler_regs.h'.
/home/nioclast0604/opentitan/util/reggen/gen_cheader.py:205: UserWarning: Cannot generate a module define of type logic [NAlerts-1:0]
  .format(param.param_type))
[99/1485] Compiling C object sw/device/lib/dif/libclkmgr_ot.a.p/dif_clkmgr.c.o
FAILED: sw/device/lib/dif/libclkmgr_ot.a.p/dif_clkmgr.c.o 
/tools/riscv/bin/riscv32-unknown-elf-clang -I/home/nioclast0604/opentitan -I/home/nioclast0604/opentitan/build-out -fcolor-diagnostics -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -g -nostdinc -isystem/home/nioclast0604/opentitan/sw/device/lib/base/freestanding -fno-asynchronous-unwind-tables -fno-common -Wimplicit-fallthrough -Wswitch-default -Wno-covered-switch-default -Wgnu -Wno-error=unused-function -Wno-unused-parameter -Wno-sign-compare -Wno-missing-field-initializers -Wno-gnu-zero-variadic-macro-arguments -Os -fvisibility=hidden -Wstrict-prototypes --gcc-toolchain=/tools/riscv -march=rv32imc -mabi=ilp32 -mcmodel=medany -MD -MQ sw/device/lib/dif/libclkmgr_ot.a.p/dif_clkmgr.c.o -MF sw/device/lib/dif/libclkmgr_ot.a.p/dif_clkmgr.c.o.d -o sw/device/lib/dif/libclkmgr_ot.a.p/dif_clkmgr.c.o -c ../sw/device/lib/dif/dif_clkmgr.c
_../sw/device/lib/dif/dif_clkmgr.c:9:10: fatal error: 'clkmgr_regs.h' file not found
#include "clkmgr_regs.h"  // Generated
         ^~~~~~~~~~~~~~~
1 error generated._
ninja: build stopped: subcommand failed.

**Comment by imphil** [comment]
What version of meson are you using? Ensure that you have installed exactly what is listed in python-requirements.txt

**Comment by nicolast0604** [comment]
0.58.0, I have trouble to install through python-requirements.txt. so I install it through sudo apt install meson.

**Comment by nicolast0604** [comment]
Thanks imphil it works now. :)

**Comment by imphil** [comment]
Yes, meson 0.58 is too new to work unfortunately.

**Comment by rswarbrick** [comment]
Looks like the problem is resolved. Closing.

**Comment by e7p** [comment]
I double checked the requirements cmake, rust, go, perl and meson (<= 0.54). However, unfortunately this did not the trick and the error of the first comment in this issue is persisting. What else should I check?

**Comment by imphil** [comment]
Since the problem originally reported was resolved for the author of this issue, can you file a new issue and let us know more about your system configuration and the error message you're seeing?

**Comment by e7p** [comment]
Of course, I just thought as it was almost exactly the same message and problem this issue could also been reopened or what. See #7407
